tools/make_patch.c

```diff
@@ -10,40 +10,67 @@
 
 struct Buffer {
 	size_t size;
-	char buffer[];
+	char data[];
 };
 
 struct Patch {
 	unsigned int offset;
 	unsigned int size;
 };
 
+struct Patches {
+	size_t size;
+	size_t capacity;
+	struct Patch *data;
+};
+
 struct Symbol {
 	struct Symbol *next;
 	int address;
 	unsigned int offset;
 	char name[];
 };
 
-void *create_buffer(void) {
-	struct Buffer *buffer = xmalloc(sizeof(struct Buffer) + 0x100);
+char *create_buffer(void) {
+	struct Buffer *buffer = xmalloc(sizeof(*buffer) + 0x100);
 	buffer->size = 0x100;
-	return buffer->buffer;
+	return buffer->data;
 }
 
-void free_buffer(void *buf) {
-	struct Buffer *buffer = (struct Buffer *)((char *)buf - sizeof(struct Buffer));
+void free_buffer(char *data) {
+	struct Buffer *buffer = (struct Buffer *)(data - sizeof(*buffer));
 	free(buffer);
 }
 
-void *expand_buffer(void *buf, const size_t size) {
-	struct Buffer *buffer = (struct Buffer *)((char *)buf - sizeof(struct Buffer));
+char *expand_buffer(char *data, size_t size) {
+	struct Buffer *buffer = (struct Buffer *)(data - sizeof(*buffer));
 	size_t old_size = buffer->size;
 	if (size > buffer->size) {
 		buffer = xrealloc(buffer, old_size + 0x100);
 	}
 	buffer->size = old_size + 0x100;
-	return buffer->buffer;
+	return buffer->data;
+}
+
+struct Patches *create_patches(void) {
+	struct Patches *patches = xmalloc(sizeof(*patches));
+	patches->size = 0;
+	patches->capacity = 0;
+	patches->data = NULL;
```

Comment:
This could start with an initial nonzero capacity, i just didn't bother.

Reply:
Resolved in https://github.com/pret/pokecrystal/pull/882/commits/630861fe91bdf4b78309b07beaa769659c56b238